src/transforms/projectionpursuit.jl

```diff
@@ -0,0 +1,199 @@
+function pindex(transform, Z, α)
+  X = rscore(Z, α)
+  I = (3/2) * mean(X)^2
+  if transform.deg > 1
+    Pⱼ₋₂, Pⱼ₋₁ = ones(size(X)), X
+    for j = 2:transform.deg
+      Pⱼ₋₂, Pⱼ₋₁ = 
+        Pⱼ₋₁, (1/j) * ((2j-1) * X .* Pⱼ₋₁ - (j-1) * Pⱼ₋₂)
+      I += ((2j+1)/2) * (mean(Pⱼ₋₁))^2
+    end
+  end
+  I
+end
+
+# projection index of the standard multivariate Gaussian
+function gindex(transform, N, q)
+  n  = transform.n
+  E  = 1.0 * I(q)
+  Iₙ = Matrix{Float64}(undef, n, q)
+  for i = 1:n-1
+    S = randn(N, q)
+    for j = 1:q
+      Iₙ[i,j] = pindex(transform, S, E[:,j])
+    end
+  end
+  Iₙ
+end
+
+function optimaxis(transform, Z)
+  _, q = size(Z)
+  E = 1.0 * I(q)
+  f(i) = pindex(transform, Z, E[:,i])
+  Iₘₐₓ, iₘₐₓ = findmax(f, 1:q)
+  E[:,iₘₐₓ]
+end
+
+function optimdiag(transform, Z, α₀)
+  q  = length(α₀)
+  E  = 1.0 * I(q)
+  Iₐ = pindex(transform, Z, α₀)
+  α  = α₀
+  for i in 1:q
+    eᵢ = E[:,i]
+    f₊ = pindex(transform, Z, (1/√(2+2α[i])) * (α + eᵢ))
+    f₋ = pindex(transform, Z, (1/√(2-2α[i])) * (α - eᵢ)) * (α[i] != 1)
+    f = (f₊ > f₋) * f₊ + (f₊ ≤ f₋) * f₋ 
+    s = (f₊ > f₋) * 1 + (f₊ ≤ f₋) * (-1)
+    if f > Iₐ
+      Iₐ = f
+      α  = (1/√(2+2s*α[i])) * (α + s*eᵢ)
+    end
+  end
+  α
+end
+
+function optimall(transform, Z, α₀)
+  f(α) = -pindex(transform, Z, α)
+  op = optimize(f, α₀)
+  minimizer(op)
+end
+
+function optim(transform, Z)
+  α₁ = optimaxis(transform, Z)
+  α₂ = optimdiag(transform, Z, α₁)
+  optimall(transform, Z, α₂)  
+end
+
+function rmstructure(transform, Z, α)
+  q = length(α)
+  Q, R = qr(hcat(α, rand(q, q-1)))
+  while norm(diag(R)) < transform.tol
+    Q, R = qr(hcat(α, rand(q, q-1)))
+  end
+  T = Z * Q
+  table = Tables.table(T)
+  newtable, colcache = apply(Quantile(), table)
+  T[:,1] = Tables.getcolumn(newtable, 1)
+  T * Q', Q, colcache
+end
+
+function stopcriteria(transform, Z, qgauss)
+  _, q = size(Z)
+  E = 1.0 * I(q)
+  Iest = [pindex(transform, Z, col) for col in eachcol(E)]
+  !(qgauss > Iest)
+end
+
+function applyfeat(transform::ProjectionPursuit, table, prep) 
+  names = Tables.columnnames(table)
+
+  trans = Quantile() → EigenAnalysis(:VDV)
+  ttable, tcache = apply(trans, table)
+  Z = Tables.matrix(ttable)
+  N, q = size(Z)
+  Igauss = gindex(transform, N, q) 
+  qgauss = Vector{Float64}(undef,q)
+  for i in 1:q
+    qgauss[i] = quantile(Igauss[:,i], transform.perc)
+  end
+  count = 0
+  cache = Tuple[]
+
+  while stopcriteria(transform, Z, qgauss)
+    α = optim(transform, Z)
+    Z, Q, colcache = rmstructure(transform, Z, α)
+    count += 1
+    if count ≥ transform.maxiter
+      @warn "Maximum number of iterations ($count) reached."
+      break
+    end
+    push!(cache,(Q,colcache))
+  end
+
+  𝒯 = (; zip(names, eachcol(Z))...)
+  newtable = 𝒯 |> Tables.materializer(table)
+  newtable, (cache,tcache)
+end
+
+function revertfeat(::ProjectionPursuit, newtable, fcache)
+  names = Tables.columnnames(newtable)
+  ppcache, tcache = fcache
+
+  coltable = newtable
```

Comment:
Let's think of a simpler name for this variable. Usually when we write a loop to update a variable, this variable name is the simplest one in the code block. The other variables, which are usually auxiliary, deserve a more specific name to guide the reader on how the main variable is being initialized and updated. So maybe rename this variable "coltable" to simply "table", and adjust the other variable names in the loop.

Reply:
What about this notation?
![image](https://user-images.githubusercontent.com/105749646/191954467-65739ab2-9962-49ed-9b62-63dd7dc2d6c2.png)
I tried a more functional notation